### ISSUE TITLE
Jump to conversation search box causes error

### ISSUE THREAD

**Issue by ReDetection** [mattermost/mattermost-mobile#735]
#### Summary
Jump to conversation search box causes error on focus, also errors on keytype 

#### Environment Information

- Device Name: iPhone 5s iOS Simulator Version 10.0 (SimulatorApp-745.10)
- OS Version: OS X 10.12.5 (16F73)
- Mattermost App Version: master, 643925469b8a04511fdceb58d2c5debcfab5f016 currently (up to date for now)
- Mattermost Server Version: 3.10.0

#### Steps to reproduce
swipe to see the sidebar, focus searchbar at the top

#### Expected behavior
keyboard appears

#### Observed behavior
[screenshot](https://user-images.githubusercontent.com/458005/28099618-c37363f6-66e7-11e7-8724-8c758b402785.png): red 'yellowbox' appears during development, saying 
```
console.error: "Unhandled promise rejection", {"line":150403,"column":22,"sourceURL":"http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false"}

error
    YellowBox.js:62:16
[...]
```
unfortunately it does not show exact line in mattermost code, can I do something to turn it on?

If dismissed, on keytype appears another error 
```
undefined is not an object (evaluating 'u.id')

<unknown>
    filtered_list.js:181:22
buildCurrentDMSForSearch
    filtered_list.js:177:75
buildSectionsForSearch
    filtered_list.js:263:37
buildData
    filtered_list.js:281:43
FilteredList
    filtered_list.js:62:39
<unknown>
    ReactCompositeComponent.js:420:30
[...]
```

**Comment by jarredwitt** [comment]
Thanks for reporting @ReDetection. @enahum @hmhealey I know what the issue is. I'll get a PR submitted this morning to fix.

**Comment by enahum** [comment]
Yup happened to me yesterday with latest beta, I was planing on taking a look but feel free to make the PR @jarredwitt

**Comment by lfbrock** [comment]
This should be fixed now

**Comment by ReDetection** [comment]
No it's not, still see an issue after #736. @lfbrock please reopen. 

[Screenshot of red yellowbox](https://user-images.githubusercontent.com/458005/28279361-72c08666-6b4a-11e7-854c-e37070a81055.png)  of unhandled promise rejection, and also [one for undefined is not an object](https://user-images.githubusercontent.com/458005/28279415-9a905400-6b4a-11e7-987c-33f6132f7a4b.png), just moved 8 lines below.
```
undefined is not an object (evaluating 'u.id')

<unknown>
    filtered_list.js:189:22
buildCurrentDMSForSearch
    filtered_list.js:185:75
buildSectionsForSearch
    filtered_list.js:271:37
[...]
```

**Comment by jarredwitt** [comment]
@ReDetection are you on the latest master or the app store release? I don't believe this fix has been pushed to production just yet.

**Comment by ReDetection** [comment]
Yep, 9db0a38e8bdd2613a97554b5dfca70360e598256 turns out to be the latest commit in master now.

Let us quickly check: I did
```
$ git pull
$ make run-ios
```
Is there anything else I should do before testing?

**Comment by ReDetection** [comment]
Can I have any feedback please? Confirm at least?

**Comment by ReDetection** [comment]
I don't admin the server but I bet they are

**Comment by ReDetection** [comment]
If you confirm the issue, I would appreciate issue reopen

**Comment by ReDetection** [comment]
I still see this after `make clean`. Please reopen or I'll make another issue.
```
undefined is not an object (evaluating 'u.id')

<unknown>
    filtered_list.js:189:22
buildCurrentDMSForSearch
    filtered_list.js:185:75
buildSectionsForSearch
    filtered_list.js:271:37
buildData
    filtered_list.js:289:43
FilteredList
    filtered_list.js:64:39
[...]
```

**Comment by jarredwitt** [comment]
@lfbrock can we reopen this? The following PR fixes this issue: https://github.com/mattermost/mattermost-mobile/pull/755

@ReDetection can you try pulling down that PR to see if it fixes the issue? If not I'm sure it will be merged into master after I fix a small piece of duplicate code.
